Kickstarter-iOS/Views/ProjectSummaryCarouselView.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+import Library
+import Prelude
+import UIKit
+
+final class ProjectSummaryCarouselView: UIView {
+  // MARK: - Properties
+
+  private lazy var collectionView: UICollectionView = {
+    UICollectionView(frame: .zero, collectionViewLayout: self.layout)
+      |> \.alwaysBounceHorizontal .~ true
+      |> \.dataSource .~ self.dataSource
+      |> \.delegate .~ self
+      |> \.showsHorizontalScrollIndicator .~ false
+  }()
+
+  private var collectionViewHeightConstraint: NSLayoutConstraint?
+
+  private let dataSource = ProjectSummaryCarouselDataSource()
+
+  private let layout: UICollectionViewFlowLayout = {
+    UICollectionViewFlowLayout()
+      |> \.minimumInteritemSpacing .~ Styles.grid(2)
+      |> \.sectionInset .~ .init(leftRight: Styles.grid(4))
+      |> \.scrollDirection .~ .horizontal
+  }()
+
+  private let viewModel: ProjectSummaryCarouselViewModelType = ProjectSummaryCarouselViewModel()
+
+  // MARK: - Lifecycle
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+
+    self.configureViews()
+    self.setupConstraints()
+    self.bindViewModel()
+  }
+
+  required init?(coder _: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - Accessors
+
+  func configure(with items: [ProjectSummaryItem]) {
+    self.viewModel.inputs.configure(with: items)
+  }
+
+  // MARK: - Configuration
+
+  private func configureViews() {
+    self.addSubview(self.collectionView)
+
+    self.collectionView.register(
```

Comment:
You can actually just call `self.collectionView.register(ProjectSummaryCarouselCell.self)` as long as your cell conforms to `ValueCell`.